### ISSUE TITLE
Warehouses only ever empty

### ISSUE THREAD

**Issue by asciant** [keallu/CSL-EmptyIt#2]
Hi @keallu,

I noticed that warehouses only ever set to empty, they are never told to fill back up again.

I dug around in the code (note that I know nothing about C#) and [I noticed this block](https://github.com/keallu/CSL-EmptyIt/blob/d6d4122e5f8e18dcc4b9dbb36ee11561ad5f63a5/EmptyIt/EmptyUtils.cs#L34-L44):

```C#
if (buildingManager.m_buildings.m_buffer[buildingId].Info.m_buildingAI is WarehouseAI)
{
    if (emptying)
    {
        buildingManager.m_buildings.m_buffer[buildingId].Info.m_buildingAI.SetEmptying(buildingId, ref buildingManager.m_buildings.m_buffer[buildingId], true);
    }
    else
    {
        buildingManager.m_buildings.m_buffer[buildingId].Info.m_buildingAI.SetFilling(buildingId, ref buildingManager.m_buildings.m_buffer[buildingId], true);
    }
}
```

It appears that the if condition on emptying is telling the Warehouse to always empty? 

The current code appears to flow like this:

1. The warehouse is currently emptying `if(emptying)` - if `emptying` is `true`.
2. The warehouse is instructed to continue emptying `SetEmptying`  - even though it is already emptying.

What I believe may have been the goal:

1. The warehouse is currently filling up `if(!emptying)` - if `emptying` is false.
2. The warehouse should be instructed to start emptying `SetEmptying`.

In short, I assume this is a bug. I do however recognise that I may have misunderstood the process that initiates this function. I assume it was called when either an upper or lower threshold is reached. Happy to close the issue if I am completely off track.
